ossc.qsf

```diff
@@ -223,11 +223,13 @@ set_global_assignment -name CYCLONEII_OPTIMIZATION_TECHNIQUE SPEED
 set_global_assignment -name ENABLE_SIGNALTAP OFF
 set_global_assignment -name USE_SIGNALTAP_FILE output_files/ossc_la.stp
 
+set_global_assignment -name AUTO_RAM_TO_LCELL_CONVERSION ON
+set_global_assignment -name MAX_RAM_BLOCKS_M4K 56
 
```

Comment:
It'd be preferred to use /* synthesis ramstyle = "logic" */; directive for small blocks that synthesis tries to infer as block RAM, e.g.
``reg [2:0] line_id_pp[1:`PP_SLGEN_PL_END-2], col_id_pp[1:`PP_SLGEN_PL_END-2] /* synthesis ramstyle = "logic" */;``

http://quartushelp.altera.com/15.0/mergedProjects/hdl/vlog/vlog_file_dir_ram.htm

Reply:
Thanks for pointing that out. I used this auto assignment to get the logic fit. My searches just pointed me to that global assignment rather than to the direct synthesis directive. I'll try that out.